lib/fluent/plugin/out_s3.rb

```diff
@@ -470,8 +474,19 @@ def setup_credentials
         credentials_options[:policy] = c.policy if c.policy
         credentials_options[:duration_seconds] = c.duration_seconds if c.duration_seconds
         credentials_options[:external_id] = c.external_id if c.external_id
-        if @s3_region
-          credentials_options[:client] = Aws::STS::Client.new(region: @s3_region)
+        credentials_options[:sts_endpoint_url] = c.sts_endpoint_url if c.sts_endpoint_url
```

Comment:
What's this? Aws::AssumeRoleCredentials.new seems not to support the parameter.
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/AssumeRoleCredentials.html

Reply:
We can see in the constructor source code for Aws::AssumeRoleCredentials.new that it accepts a customizable set of options parameters. We can also see that if the option is not in the set of assume_role_options then it will go ahead and set those options to the STS::Clinet constructor. this effectively forwards the STS client parameters for us.

The underlying AWS client and API for STS will take care of the rest for us.

for more reference please refer to awslabs solution in https://github.com/awslabs/aws-fluent-plugin-kinesis/blob/47ad650a907c6524be323b66860e4757fb02ddd9/lib/fluent/plugin/kinesis_helper/client.rb#L117

Reply:
https://github.com/awslabs/aws-fluent-plugin-kinesis/blob/47ad650a907c6524be323b66860e4757fb02ddd9/lib/fluent/plugin/kinesis_helper/client.rb#L47

awslabs defines `sts_endpoint_url` here so it can access it. but you don't. it causes CI fail.
Maybe you should define in `assume_role_credentials`  not in `web_identity_credentials ` .
https://github.com/fluent/fluent-plugin-s3/blob/b3210b791f1b4df36853986a8e1068f40fe61f9f/lib/fluent/plugin/out_s3.rb#L31-L44